### PR TITLE
[eslint config][major][guide] Always require parens around arrow function arguments

### PR DESCRIPTION
**Pull request by sharmilajesupaul**
guide + eslint config change to always require parens around arrow function arguments.

### CODE REVIEW

**Comment by olstenlarck** [comment]
Yeaaaah!! :tada: :)

**Comment by ljharb** [comment]
Thanks - since this is a semver-major, I'm going to hold off on merging it until after #1834 is released.

File: README.md [hunk 1]
```diff
@@ -1018,22 +1021,27 @@ Other Style Guides
     ```
 
   <a name="arrows--one-arg-parens"></a><a name="8.4"></a>
-  - [8.4](#arrows--one-arg-parens) If your function takes a single argument and doesn’t use braces, omit the parentheses. Otherwise, always include parentheses around arguments for clarity and consistency. Note: it is also acceptable to always use parentheses, in which case use the [“always” option](https://eslint.org/docs/rules/arrow-parens#always) for eslint. eslint: [`arrow-parens`](https://eslint.org/docs/rules/arrow-parens.html)
+  - [8.4](#arrows--one-arg-parens) Always include parentheses around arguments for clarity and consistency. eslint: [`arrow-parens`](https://eslint.org/docs/rules/arrow-parens.html)
 
-    > Why? Less visual clutter.
+    > Why? Minimizes diff churn when adding or removing arguments.
```
Comment (lencioni) [inline README.md]: This is a much better reason!

**Comment by alexzherdev** [comment]
Since #1834 is done now, can this be rebased and merged?

File: README.md [hunk 2]
```diff
@@ -1018,22 +1021,27 @@ Other Style Guides
...
 
     ```javascript
     // bad
-    [1, 2, 3].map((x) => x * x);
+    [1, 2, 3].map((x => x * x);
```
Comment (j-f1) [inline README.md]: Looks like the `(` should be deleted here:

```suggestion
	    [1, 2, 3].map(x => x * x);
```

Comment (ljharb) [reply README.md]: whoops, this should actually be *adding* the `)`.

**Comment by tunnckoCore** [comment]
Sweeet. One more rule which I like and forced in my preset, got merged. :)
So I guess, I'm not that bad at decisions and logical & good style ;p

**Comment by elyobo** [comment]
Overriding this

> Why? Minimizes diff churn when adding or removing arguments

I don't find this to be the case; there's still a diff on the same line, so doesn't really make any meaningful change to the diffs, but :point_down: is still meaningful.

> Why? Less visual clutter. 

Glad that it makes some people happy though, and I don't mind overriding things if necessary :)

**Pull request merged**